core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/wrapper/data/RdfsSubClassOfReasoner.java

```diff
@@ -149,27 +150,48 @@ private Set<Resource> resolveTypes(Resource value, Map<Resource, Set<Resource>>
 		return iris != null ? iris : Collections.emptySet();
 	}
 
-	public static RdfsSubClassOfReasoner createReasoner(ShaclSailConnection shaclSailConnection) {
+	public static RdfsSubClassOfReasoner createReasoner(SailConnection sailConnection,
+			ValidationSettings validationSettings) {
+		return createReasoner(sailConnection, null, validationSettings);
+	}
+
+	public static RdfsSubClassOfReasoner createReasoner(SailConnection sailConnection, SailConnection secondConnection,
+			ValidationSettings validationSettings) {
 		long before = 0;
-		if (shaclSailConnection.sail.isPerformanceLogging()) {
+		if (validationSettings.isPerformanceLogging()) {
```

Comment:
One fairly large part of this PR was to eliminate the use of the ShaclSail or ShaclSailConnection in the validation code. So I've introduced some simple config classes to pass in settings that would otherwise be available directly from the ShaclSail or ShaclSailConnection.